File: lib/src/main/java/com/github/util/StringUtils.java
package com.github.util;

import androidx.annotation.Nullable;

/**
 * String utilities.
 *
 * @author Moshe Waisberg
 */
public class StringUtils {

    private StringUtils() {
    }

    @Nullable
    public static String toString(CharSequence s) {
        return (s != null) ? s.toString() : null;
    }
}

File: lib/src/main/java/com/github/util/StringUtils.kt
package com.github.util

/**
 * String utilities.
 *
 * @author Moshe Waisberg
 */
object StringUtils {
    @JvmStatic
    fun toString(s: CharSequence?): String? {
        return s?.toString()
    }

    @JvmStatic
    fun compareTo(s1: String?, s2: String?, ignoreCase: Boolean = false): Int {
        if (s1 == s2) return 0
        if (s1 == null) return -1
        if (s2 == null) return +1
        return s1.compareTo(s2, ignoreCase = ignoreCase)
    }
}
